lib/broccoli/translation-reducer.js

```diff
@@ -80,12 +80,22 @@ class TranslationReducer extends CachingWriter {
     }
   }
 
-  findMissingKeys(target, defaultTranslationKeys, locale) {
-    let targetProps = propKeys(target);
+  findMissingTranslations(translations) {
+    const locales = Object.keys(translations);
 
-    defaultTranslationKeys.forEach(property => {
-      if (targetProps.indexOf(property) === -1) {
-        this._logLocale(locale, `"${property}" message is missing`);
+    // [['de',['foo.bar', 'bar.baz'], ['en',['foo.baz','bar.baz']]
+    const localeKeys = locales.map(locale => [locale, propKeys(translations[locale])]);
+
+    // create flattened list of all unique translation keys
+    const allTranslationKeys = Array.from(new Set(Array.prototype.concat(...localeKeys.map(([, keys]) => keys))));
```

Comment:
Do not need to convert back to an array, can stay a Set

Reply:
Thanks, I've changed it.
Forgot that `Set` has a tiny amount of array methods built-in.